stock_maturity/views/stock_lot_views.xml

```diff
@@ -0,0 +1,60 @@
+<?xml version="1.0" encoding='UTF-8' ?>
+<odoo>
+    <record id="view_production_lot_form" model="ir.ui.view">
+        <field name="name">stock.production.lot.inherit.form</field>
+        <field name="model">stock.lot</field>
+        <field name="inherit_id" ref="stock.view_production_lot_form" />
+        <field name="arch" type="xml">
+            <field name="ref" position="after">
+                <field name="maturity_date" />
+            </field>
+            <xpath expr="//div[hasclass('oe_title')]" position="inside">
+                <span
+                    class="badge text-bg-warning"
+                    invisible="not product_maturity_alert"
+                >Not mature yet</span>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="view_production_lot_tree" model="ir.ui.view">
```

Comment:
Would be great to add a search view that adds "Already Matured / Not Matured yet" to filters.